quick_select/QuickSelect.java

```diff
@@ -0,0 +1,64 @@
+public class QuickSelect {
+    /*
+    * partition function
+    * array     : array on which partitioning has to be done
+    * left      : left index of the partitioning subarray
+    * right     : right index of the partitioning subarray
+    * pivotIndex : pivot index from which partition has to done
+    * return    : the index of the last element of the left subarray
+     */
+    private static int partition(int[] array, int left, int right, int pivotIndex) {
+        int pivotValue = array[pivotIndex];
+        int temp = array[right];
+        array[right] = array[pivotIndex];
+        array[pivotIndex] = temp;
+        int storeIndex = left;
+        while (left < right) {
+            if (array[left] < pivotValue) {
+                temp = array[storeIndex];
+                array[storeIndex] = array[left];
+                array[left] = temp;
+                storeIndex++;
+            }
+            left++;
+        }
+        temp = array[right];
+        array[right] = array[storeIndex];
+        array[storeIndex] = temp;
+        return storeIndex;
+    }
+
+    /*
+    * Quick Select function
+    * left        : left index of the subarray
+    * right       : right index of the subarray
+    * pos         : position to find the element using quick sort
+    * return      : the value of element at pos place in the sorted array
+     */
+    public static int quickSelect(int[] array, int left, int right, int pos) {
+        int pivotIndex;
+        if(pos < 0 || pos >= array.length) {
+            throw new ArithmeticException("ArrayIndexOutOfBoundException");
```

Comment:
I think you got it wrong. I was saying `throw new IndexOutOfBoundsException("index: " + pos);'`